src/omotes_rest/postgres_interface.py

```diff
@@ -0,0 +1,335 @@
+            },
+        )
+    except Exception as e:
+        logger.error(e)
+
+    # Bind the global session to the actual engine.
+    Session.configure(bind=engine)
+
+    return engine
+
+
+class PostgresInterface:
+    """Interface to the SQL database for any queries to persist or retrieve job information.
+
+    Note: The interface may be called from many threads at once. Therefor each query/function
+    in this interface must set up a Session (scope) separately.
+    """
+
+    db_config: PostgreSQLConfig
+    """Configuration on how to connect to the database."""
+    engine: Engine
+    """Engine for starting connections to the database."""
+
+    def __init__(self, postgres_config: PostgreSQLConfig) -> None:
+        """Create the PostgreSQL interface."""
+        self.db_config = postgres_config
+
+    def start(self) -> None:
+        """Start the interface and connect to the database."""
+        self.engine = initialize_db("omotes_rest", self.db_config)
+
+    def stop(self) -> None:
+        """Stop the interface and dispose of any connections."""
+        if self.engine:
+            self.engine.dispose()
+
+    def put_new_job(
+            self,
+            job_id: uuid4,
+            job_input: JobInput
+    ) -> None:
+        """Insert a new job into the database.
+
+        Note: Assumption is that the job_id is unique and has not yet been added to the database.
+
+        :param job_id: Unique identifier of the job.
+        :param job_input: Received input for the job
+        """
+        with session_scope(do_expunge=False) as session:
+            new_job = JobRest(
+                job_id=job_id,
+                job_name=job_input.job_name,
+                workflow_type=job_input.workflow_type,
+                status=JobRestStatus.REGISTERED,
+                progress_fraction=0,
+                progress_message="Job registered.",
+                registered_at=datetime.now(),
+                timeout_after_s=job_input.timeout_after_s,
+                user_name=job_input.user_name,
+                project_name=job_input.project_name,
+                input_params_dict=job_input.input_params_dict,
+                input_esdl=job_input.input_esdl
+            )
+            session.add(new_job)
+        logger.debug("Job %s is submitted as new job in database", job_id)
+
+    def set_job_status(self, job_id: uuid4, new_status: int, result: JobResult = None) -> None:
+        """Set the status of the job to 'SUBMITTED'.
+
+        :param job_id: Job to set the status to 'SUBMITTED'.
+        :param new_status: new status 'int' identifier from protobuf message.
+        :param result: optional result containing the result type, logs and output esdl.
+        """
+        logger.debug("For job '%s' received new status '%s'", job_id, new_status)
+
+        with session_scope() as session:
+            if new_status == 0:  # REGISTERED
```

Comment:
1. Perhaps use `omotes_sdk.omotes_sdk_protocol.job_pb2.JobStatusUpdate.JobStatus.REGISTERED` instead of hardcoded values? 
2. I would advise to move this mapping out of `postgres_interface.py` to `RestInterface`. The Postgres interface should not care about `JobResult` and other SDK-specific classes. I would expect only the logic to set a new status here.